perf_dashboard/assets/js/cur_alert.js

```diff
@@ -1,79 +1,239 @@
 window.onload = function () {
-  let chart_cur_pattern_mixer_serveronly = [];
-  let chart_cur_pattern_mixer_both = [];
-  let chart_cur_pattern_nomixer_serveronly = [];
-  let chart_cur_pattern_nomixer_both = [];
-  let chart_cur_pattern_v2_serveronly = [];
-  let chart_cur_pattern_v2_both = [];
-
-  for (i = 0; i < cur_pattern_mixer_serveronly.length; i++) {
-    if(cur_pattern_mixer_serveronly[i][3] === "null") {
-      chart_cur_pattern_mixer_serveronly.push({ x: new Date(cur_pattern_mixer_serveronly[i][0],
-              cur_pattern_mixer_serveronly[i][1]-1, cur_pattern_mixer_serveronly[i][2]),
+  // p90
+  let chart_cur_pattern_mixer_serveronly_p90 = [];
+  let chart_cur_pattern_mixer_both_p90 = [];
+  let chart_cur_pattern_nomixer_serveronly_p90 = [];
+  let chart_cur_pattern_nomixer_both_p90 = [];
+  let chart_cur_pattern_v2_serveronly_p90 = [];
+  let chart_cur_pattern_v2_both_p90 = [];
+
+  for (i = 0; i < cur_pattern_mixer_serveronly_p90.length; i++) {
+    if(cur_pattern_mixer_serveronly_p90[i][3] === "null") {
+      chart_cur_pattern_mixer_serveronly_p90.push({ x: new Date(cur_pattern_mixer_serveronly_p90[i][0],
+              cur_pattern_mixer_serveronly_p90[i][1]-1, cur_pattern_mixer_serveronly_p90[i][2]),
+          y: null })
+    } else {
+      chart_cur_pattern_mixer_serveronly_p90.push({ x: new Date(cur_pattern_mixer_serveronly_p90[i][0],
+              cur_pattern_mixer_serveronly_p90[i][1]-1, cur_pattern_mixer_serveronly_p90[i][2]),
+          y: cur_pattern_mixer_serveronly_p90[i][3]- cur_pattern_mixer_base_p90[i][3]})}
+  }
+
+  for (i = 0; i < cur_pattern_mixer_both_p90.length; i++) {
+    if(cur_pattern_mixer_both_p90[i][3] === "null") {
+      chart_cur_pattern_mixer_both_p90.push({ x: new Date(cur_pattern_mixer_both_p90[i][0],
+              cur_pattern_mixer_both_p90[i][1]-1, cur_pattern_mixer_both_p90[i][2]),
+          y: null })
+    } else {
+      chart_cur_pattern_mixer_both_p90.push({ x: new Date(cur_pattern_mixer_both_p90[i][0],
+              cur_pattern_mixer_both_p90[i][1]-1, cur_pattern_mixer_both_p90[i][2]),
+          y: cur_pattern_mixer_both_p90[i][3]- cur_pattern_mixer_base_p90[i][3]})}
+  }
+
+  for (i = 0; i < cur_pattern_nomixer_serveronly_p90.length; i++) {
```

Comment:
can we make this more parameterized so we don't have so much copy and pasting? I would expect adding p99 to be 1-2 lines of code adding p99 to a list

Reply:
oh, yes, we can, let me refactor this 😶